From 54e0d99710e3ce2b31f18b3a4a0dc1e62300d3e6 Mon Sep 17 00:00:00 2001
From: "Zhou, Furong" <furong.zhou@intel.com>
Date: Wed, 28 Aug 2024 14:39:59 +0800
Subject: [PATCH 05/76] Enable/Disable more RT kernel config

Update RT kernel configuration to:
1. Reduce task scheduling-clock overhead
2. Disable CPU governor Linux features
3. Disable Linux power management runtime features
4. Enable more accurate task and CPU time accounting
5. Enable more preemptive task scheduling policies and CPU
   temporal isolation

Signed-off-by: Zhou, Furong <furong.zhou@intel.com>
---
 Intel/config-rt | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/Intel/config-rt b/Intel/config-rt
index f19e2b402c7ec..0ac6e69d4aa22 100644
--- a/Intel/config-rt
+++ b/Intel/config-rt
@@ -100,7 +100,7 @@ CONFIG_NO_HZ_COMMON=y
 # CONFIG_HZ_PERIODIC is not set
 # CONFIG_NO_HZ_IDLE is not set
 CONFIG_NO_HZ_FULL=y
-CONFIG_NO_HZ=y
+# CONFIG_NO_HZ is not set
 CONFIG_HIGH_RES_TIMERS=y
 CONFIG_CLOCKSOURCE_WATCHDOG_MAX_SKEW_US=100
 # end of Timers subsystem
@@ -485,11 +485,11 @@ CONFIG_EFI_HANDOVER_PROTOCOL=y
 # CONFIG_EFI_MIXED is not set
 # CONFIG_EFI_FAKE_MEMMAP is not set
 CONFIG_EFI_RUNTIME_MAP=y
-# CONFIG_HZ_100 is not set
+# CONFIG_HZ_1000 is not set
 # CONFIG_HZ_250 is not set
 # CONFIG_HZ_300 is not set
-CONFIG_HZ_1000=y
-CONFIG_HZ=1000
+CONFIG_HZ_100=y
+CONFIG_HZ=100
 CONFIG_SCHED_HRTICK=y
 CONFIG_ARCH_SUPPORTS_KEXEC=y
 CONFIG_ARCH_SUPPORTS_KEXEC_FILE=y
@@ -548,9 +548,7 @@ CONFIG_ARCH_HAS_ADD_PAGES=y
 # Power management and ACPI options
 #
 CONFIG_ARCH_HIBERNATION_HEADER=y
-CONFIG_SUSPEND=y
-CONFIG_SUSPEND_FREEZER=y
-# CONFIG_SUSPEND_SKIP_SYNC is not set
+# CONFIG_SUSPEND is not set
 CONFIG_HIBERNATE_CALLBACKS=y
 CONFIG_HIBERNATION=y
 CONFIG_HIBERNATION_SNAPSHOT_DEV=y
@@ -643,7 +641,7 @@ CONFIG_CPU_FREQ_DEFAULT_GOV_PERFORMANCE=y
 CONFIG_CPU_FREQ_GOV_PERFORMANCE=y
 CONFIG_CPU_FREQ_GOV_POWERSAVE=m
 CONFIG_CPU_FREQ_GOV_USERSPACE=m
-CONFIG_CPU_FREQ_GOV_ONDEMAND=m
+# CONFIG_CPU_FREQ_GOV_ONDEMAND is not set
 CONFIG_CPU_FREQ_GOV_CONSERVATIVE=m
 CONFIG_CPU_FREQ_GOV_SCHEDUTIL=y
 
-- 
2.43.0

